### DISCUSSION TITLE
Header case conversion

### DISCUSSION THREAD

**Discussion by nuanxinqing123** [go-resty/resty#911] [Q&A] [answered]
I came across an API interface where I Header setting
{
‘ts’: ‘1720943886093’
}
It gets automatically converted to
Ts: 1720943886093
But the interface is case-sensitive, and can only fetch lowercase (ts).

Is there any way to avoid the automatic case conversion?

**Comment by jeevatkm** [comment, marked as answer]
@nuanxinqing123 Go lang makes the automatic conversation of the header key into Canonical format. You could read about that online.

In Resty, you could use these methods to preserve the header key format as-is -
* https://pkg.go.dev/github.com/go-resty/resty/v2#Client.SetHeaderVerbatim
* https://pkg.go.dev/github.com/go-resty/resty/v2#Request.SetHeaderVerbatim
